sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -76,17 +88,26 @@ public DigitalTwinsClientBuilder()
         httpLogOptions = new HttpLogOptions();
     }
 
-    private static HttpPipeline buildPipeline(TokenCredential tokenCredential, String endpoint,
-                                              HttpLogOptions httpLogOptions, HttpClient httpClient,
-                                              List<HttpPipelinePolicy> additionalPolicies, RetryPolicy retryPolicy,
-                                              Configuration configuration, Map<String, String> properties) {
+    private static HttpPipeline buildPipeline(TokenCredential tokenCredential,
+                                              String endpoint,
+                                              HttpLogOptions httpLogOptions,
+                                              ClientOptions clientOptions,
+                                              HttpClient httpClient,
+                                              List<HttpPipelinePolicy> additionalPolicies,
+                                              RetryPolicy retryPolicy,
+                                              Configuration configuration,
+                                              Map<String, String> properties) {
         // Closest to API goes first, closest to wire goes last.
         List<HttpPipelinePolicy> policies = new ArrayList<>();
 
         String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
         String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
 
-        policies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion, configuration));
+        // Give precedence to applicationId configured in clientOptions over the one configured in httpLogOptions.
+        // Azure.Core deprecated setting the applicationId in httpLogOptions.
+        String applicationId =
```

Comment:
How do we feel about formatting ternary operations like we do in .NET?

```java
String applicationId = clientOptions == null
    ? httpLogOptions.getApplicationId()
    : clientOptions.GetApplicationId();
```

Reply:
Sure, that looks a bit cleaner